common/djangoapps/third_party_auth/tests/specs/test_testshib.py

```diff
@@ -488,6 +488,57 @@ def test_register_sapsf_metadata_present(self):
         )
         self._test_register(country=expected_country)
 
+    def test_register_sapsf_with_value_default(self):
```

Comment:
is it possible to write a similar test case for non SAPSF SAML IdPs?

Reply:
I tried writing a test case for testshib, but even after some refactoring, I am not able to write a test that passes. The main problem is getting a sample response that is missing the email field. I keep getting this error `Signature validation failed. SAML Response rejected`. I tried signing my custom response using https://www.samltool.com/sign_response.php but for some unknown reason, it is not working. 
After discussing with @georgebabey I have decided to skip this test case for now. I have verified testshib manually though and it is working as expected.